plot-lib/plot/private/plot2d/plot-area.rkt

```diff
@@ -337,8 +337,16 @@
     (: draw-y-far-tick-labels? Boolean)
     (define draw-x-tick-labels? (plot-x-tick-labels?))
     (define draw-y-tick-labels? (plot-y-tick-labels?))
-    (define draw-x-far-tick-labels? (or (plot-x-far-tick-labels?) (not (and (plot-x-axis?) (equal? x-ticks x-far-ticks)))))
-    (define draw-y-far-tick-labels? (or (plot-y-far-tick-labels?) (not (and (plot-y-axis?) (equal? y-ticks y-far-ticks)))))
+    (define draw-x-far-tick-labels?
+      (let ((flag (plot-x-far-tick-labels?)))
+        (if (eq? flag 'auto)
+            (not (and (plot-x-axis?) (equal? x-ticks x-far-ticks)))
+            flag)))
```

Comment:
Maybe replace this code with a function call `(f (plot-x-far-tick-labels?) (plot-x-axis?) x-ticks x-far-ticks)` ?

I can't think of what to name `f` though, so maybe not.